File: app/src/main/java/com/healios/dreams/ui/dashboard/home/DashboardNonCompletedChallengesRecyclerViewAdapter.java
package com.healios.dreams.ui.dashboard.home;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.healios.dreams.R;
import com.healios.dreams.databinding.ItemDashboardNonCompletedChallengeBinding;

public class DashboardNonCompletedChallengesRecyclerViewAdapter extends RecyclerView.Adapter<DashboardNonCompletedChallengesRecyclerViewAdapter.DashboardNonCompletedChallengesViewHolder> {
    //TODO:


    @NonNull
    @Override
    public DashboardNonCompletedChallengesViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_dashboard_non_completed_challenge, parent, false);
        return new DashboardNonCompletedChallengesViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull DashboardNonCompletedChallengesViewHolder holder, int position) {

        holder.bind();
    }

    @Override
    public int getItemCount() {
        return 0;
    }

    class DashboardNonCompletedChallengesViewHolder extends RecyclerView.ViewHolder {

        private ItemDashboardNonCompletedChallengeBinding binding;

        public DashboardNonCompletedChallengesViewHolder(@NonNull View itemView) {
            super(itemView);
            binding = ItemDashboardNonCompletedChallengeBinding.bind(itemView);

        }

        public void bind() {

        }
    }
}

File: app/src/main/java/com/healios/dreams/ui/dashboard/home/DashboardNonCompletedChallengesRecyclerViewAdapter.kt
package com.healios.dreams.ui.dashboard.home

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.healios.dreams.databinding.ItemDashboardNonCompletedChallengeBinding
import com.healios.dreams.model.Test

class DashboardNonCompletedChallengesRecyclerViewAdapter :
    RecyclerView.Adapter<DashboardNonCompletedChallengesRecyclerViewAdapter.DashboardNonCompletedChallengesViewHolder>() {

    private var nonCompletedChallenges: List<Test>? = ArrayList(5)

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): DashboardNonCompletedChallengesViewHolder {
        val binding = ItemDashboardNonCompletedChallengeBinding.inflate(
            LayoutInflater.from(parent.context),
            parent,
            false
        )
        return DashboardNonCompletedChallengesViewHolder(binding)
    }

    override fun getItemCount(): Int {
        nonCompletedChallenges?.let {
            //FIXME: Uncomment
            return 5//it.size
        }
        return 0
    }

    override fun onBindViewHolder(
        holder: DashboardNonCompletedChallengesViewHolder,
        position: Int
    ) {
        //val test = nonCompletedChallenges!![position]
        //holder.bind(test)
    }


    fun setData(nonCompletedChallengesList: List<Test>?) {
        nonCompletedChallenges = nonCompletedChallengesList
        notifyDataSetChanged()
    }

    //region: ViewHolder
    inner class DashboardNonCompletedChallengesViewHolder(var binding: ItemDashboardNonCompletedChallengeBinding) :
        RecyclerView.ViewHolder(binding.root) {

        fun bind(test: Test?) {
            //TODO:
            with(binding) {

            }
        }
    }
    //endregion
}